model/task_lifecycle.go

```diff
@@ -31,109 +31,114 @@ type StatusChanges struct {
 	BuildComplete    bool
 }
 
-func SetActiveState(t *task.Task, caller string, active bool) error {
-	originalTasks := []task.Task{*t}
-	if t.DisplayOnly {
-		execTasks, err := task.Find(task.ByIds(t.ExecutionTasks))
-		if err != nil {
-			return errors.Wrapf(err, "error getting execution tasks")
-		}
-		originalTasks = append(originalTasks, execTasks...)
-	}
-
-	if active {
-		// if the task is being activated and it doesn't override its dependencies
-		// activate the task's dependencies as well
-		tasksToActivate := []task.Task{}
-		if !t.OverrideDependencies {
-			deps, err := task.GetRecursiveDependenciesUp(originalTasks, nil)
+func SetActiveState(caller string, active bool, tasks ...*task.Task) error {
+	tasksToActivate := []task.Task{}
+	versionsToActivate := []Version{}
+	for _, t := range tasks {
+		originalTasks := []task.Task{*t}
+		originalTasks = append(originalTasks, *t)
+		if t.DisplayOnly {
+			execTasks, err := task.Find(task.ByIds(t.ExecutionTasks))
 			if err != nil {
-				return errors.Wrapf(err, "error getting tasks '%s' depends on", t.Id)
+				return errors.Wrapf(err, "error getting execution tasks")
```

Comment:
same question about using a catcher

Reply:
Sgtm, since if it failed halfway through the loop there would be a subset of activated tasks anyway. Would likely be a bit more resilient.